### ISSUE TITLE
issue in training !

### ISSUE THREAD

**Issue by 008karan** [ecthros/labelReader#8]
when I ran training by this command: `./darknet detector train data/obj.data yolo-obj.cfg darknet53.conv.74 I got this message 
`
`Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Resizing
320 x 320 

 Wrong annotation: class_id = 2. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 5. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 3. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 1. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 1. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 2. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 4. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 2. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 3. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 3. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 4. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 4. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 4. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 1. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 1. But class_id should be [from 0 to 1] 

 Wrong annotation: class_id = 3. But class_id should be [from 0 to 1] 

and also a bad_label.list file. I annotated data using vott only. Dont know whats going wrong.
 Inside bad_label.list file there this lines.
`data/obj/pan83.txt Wrong annotation: class_id = 2. But class_id should be [from 0 to 1]`

every time I run training I get this error but for different **.txt** file names in bad_label.list file. 
How get rid of this.. I clonned your repo then followed darknet instruction for training to detect custom objects.
Do I need to compile darknet first or just follow instruction for training

**Comment by ecthros** [comment]
Are you classifying multiple objects? If so, you'll need to change the parameters in the yolo config file.

**Comment by 008karan** [comment]
Yes I have six classes. I have changed parameters  according to the darknet repo.

**Comment by 008karan** [comment]
ok now its training. Can you tell for how many iteration I should train? From past 1 hr its been 4 iterations only.

**Comment by 008karan** [comment]
I trained my network. Can you explain how to get ocr after training the network? I was following the labelReader.py code and replaced .cfg file, weights file ,data file but when I run I get this message:
Using TensorFlow backend. [+] Initializing Darknet [/] Init Classifier failed, exiting now
please help!

**Comment by valerosha** [comment]
Hi) @008karan   How did you solve this problem?

**Comment by 008karan** [comment]
you need to edit your yolo config file. Follow this repo https://github.com/AlexeyAB to train your classifier darknet.

**Comment by Aveen1** [comment]
I'm having the same issue, can someone share solution of it? I've changed my cfg file and.names file but error is still there.

**Comment by windmewest** [comment]
> I'm having the same issue, can someone share solution of it? I've changed my cfg file and.names file but error is still there.

Hi, I got the same issue. And I solve it by check the bad.list file in the same content as darknet.exe. Maybe you could see some class id in the bad.list is out of range.

**Comment by Aveen1** [comment]
Thanks but I was having this error due to same number 0 as class given to both classes in .txt files which were converted from xml to txt. I change them to 0= Reading, 1= Texting in text files and it worked fine.
I was working on these 2 classes. So check class id's if you are facing this error.

**Comment by Borahb** [comment]
I am also facing this issue
[+] Initializing Darknet
[/] Init Classifier failed, exiting now

Anyone have a solution, please suggest.

**Comment by Sivabanu** [comment]
I also facing this problem , Any one give a solution 
Is fault in cfg file or image labling ??
